bc-madr-specification.md

```diff
@@ -0,0 +1,196 @@
+
+## 2. Pillars of BC-MADRs
+
+BC-MADRs and their supporting systems MUST fulfill several fundamental pillars:
+
+### 2.1 Storage and proximity
+
+1. For any project (source code or otherwise), BC-MADRs MUST be stored in a git repository.
+
+2. If a project is software development, BC-MADRs MUST be stored in the same repository as the source code.
+
+3. In a repository, BC-MADRs MUST be stored at the following path and nowhere else:
+
+	`<repo_root>/docs/decisions`
+
+	⚠TBD: Should we allow decision subfolders or require all decisions be kept in the same folder?
+
+4. When naming BC-MADR files, you are:
+
+	- REQUIRED to use kebab-naming. (`this-is-kebab-naming`)
+
+	- REQUIRED to save as a Markdown file. (`this-is-kebab-naming.md`)
+
+	- REQUIRED Use an incremental scheme in the filename.
+
+	- RECOMMENDED to use an integer auto-increment scheme, e.g.
+
+		```bash
+		0001-this-is-kebab-naming.md
+		0002-this-is-more-kebab-naming.md
+		0003-this-is-additional-kebab-naming.md
+		```
+		
+		Alternatively, you can use whatever incremental scheme that fits best for your project, but: when the filenames are sorted in ascending order, the list of BC-MADRs MUST be ordered from oldest to newest.
+
+	- RECOMMENDED to write the same title as in the file itself, so it is easier to find decisions. Example:
+
+		`0001-use-the-bc-madr-decision-format-in-bc-gov.md`:
+
+		```markdown
+		[//]: # (bc-madr v0.1 )
+
+		# Use the BC-MADR decision format in BC Gov
+
+		...
+		```
+
+### 2.2 Dedicated and decoupled history
+
+A given set of BC-MADRs MUST only belong to a single project in order to maintain a single source of truth.
+
+All significant decisions made for a project's technical direction MUST be recorded in a BC-MADR.
+
+BC-MADRs MUST NOT contain direct references to the project's source code, as these references are not enforced by tools like compilers. 
+
+BC-MADRs MAY contain code examples if they are relevant to a decision and are not intended as a direct reference to project source code.
+
+#### 2.2.1 Significant decisions
+
+If the answer to any of the following questions is "Yes," create an ADR to record the decision.
+
+- Does this decision affect or change the architecture of the project?
+
+	- Integrating with APIs, databases, platforms, etc.
+
+- Do I need to make a pros/cons list for this decision?
+
+- Does this decision affect the User Experience (UX) of the project?
+
+	- Adding, changing, or removing features
+
+	- Cosmetic changes
+
+- Am I implementing a feature with logic that is not obvious to a user?
+
+- If someone asks "why" about a decision related to the project, does it require deep understanding of that project?
+
+#### 2.2.2 Technical direction
+
+⚠TBD: Should BC-MADRs be used only for technical decisions, or also include operational decisions?
```

Comment:
Thoughts?

Reply:
In NRM we tried to come up with and publish a definition of "Architecturally Significant Decisions" - which was more about when to engage with your architect, but I could see something similar in this regard to help guide teams when an ADR is needed.  We adopted something called ARD (architecture response document) and ADR (architecture decision record), appears to be a hybrid of this that's modeled after the Nygard word.  I like the 'as code beside code' model as it allows traceability beyond forks.

https://apps.nrs.gov.bc.ca/int/confluence/display/AR/Architecturally+Significant+Decisions

https://apps.nrs.gov.bc.ca/int/confluence/pages/viewpage.action?pageId=65672124

https://apps.nrs.gov.bc.ca/int/confluence/display/AR/Architecture+Decision+Records

Reply:
TBD in v0.2, will open a discussion thread on this.